doc: Update known issues for 150 release (#3771)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

  The :ref:`asset_tracker` application does not wait for connection to nRF Connect for Cloud before trying to send data.
  This causes the application to crash if the user toggles one of the switches before the kit is connected to the cloud.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0

Reviewer: Is fixed in 1.5.0

NRF91-989: Unable to bootstrap after changing SIMs
  In some cases, swapping the SIM card may trigger the bootstrap Pre-Shared Key to be deleted from the device. This can prevent future bootstraps from succeeding.

.. rst-class:: v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0

NCSDK-5666: LTE Sensor Gateway
  The :ref:`lte_sensor_gateway` sample crashes when Thingy:52 is flipped.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

Reviewer: is fixed in 1.5.0

nRF5340
=======

.. rst-class:: v1-5-0

802.15.4 Service Layer (SL) library support for the nRF53

>   The nRF53 support in the 802.15.4 Service Layer (SL) library is available only in the open-source version of the library.

Reviewer 2: ```suggestion
  The binary variant of the  802.15.4 Service Layer (SL) library for nRF53  does not support such features as synchronization of TIMER with RTC, timestamping of received frames. Because of that such 802.15.4 features as delayed transmission or delayed reception are note available for nRF53.
```

Reviewer 3: Added with reference to the provided JIRA issue.

Thread
======

.. rst-class:: v1-5-0

> KRKNWK-8286: Faulty UART responsiveness in Thread NCP sample

Reviewer 4: This was only affecting 1.4.99. Fixed for 1.5.

Reviewer 3: Removed the whole issue, as v1.5.0 replaces v1.4.99-dev1.

  A race condition in the UARTE driver causes the retrieval of Spinel responses to fail sometimes for the :ref:`ot_ncp_sample` sample.

  **Workaround:** Build the sample with :option:`CONFIG_UART_0_ENHANCED_POLL_OUT` set to ``n`` or cherry-pick the upstream `Zephyr commit 2db49c`_.

.. rst-class:: v1-5-0

> KRKNWK-8262: CoAP Client Sample crashes for nRF5340

Reviewer 4: This was only affecting 1.4.99. Fixed for 1.5.

Reviewer 3: Removed the whole issue, as v1.5.0 replaces v1.4.99-dev1.

  **Workaround:** Use the application default thread, a dedicated workqueue, or a new thread to handle operations that require radio interactions.
  If you cannot avoid using the system workqueue, cherry-pick the upstream `Zephyr commit f12536`_.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

KRKNWK-6848: Reduced throughput
  Performance testing for :ref:`NCP sample <ot_ncp_sample>` shows a decrease of throughput of around 10-20% compared with the standard OpenThread.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

> KRKNWK-7885: Throughput is lower when using CC310 nrf_security backend

Reviewer 4: Fixed in 1.5

Reviewer 3: Removed the latest tag.

  * :option:`CONFIG_OBERON_MBEDTLS_CCM_C` to ``y``
  * :option:`CONFIG_CC3XX_MBEDTLS_AES_C` to ``n``

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

> KRKNWK-7721: MAC counter updating issue

Reviewer 4: Fixed in 1.5

Reviewer 3: Removed the latest tag.

Zigbee
======

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 5: Fixed in 1.5.0

nRF Desktop
===========

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0

Reviewer 3: @MarekPieta , any update on this?

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0

Reviewer 6: The issue was fixed for v1.5.0 (I even verified that it works fine). We should not add this tag.

  **Workaround:** Set :option:`CONFIG_BT_L2CAP_TX_BUF_COUNT` >= :option:`CONFIG_BT_ATT_TX_MAX` + 2.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

> NCSDK-6845: Pairing failure with simultaneous pairing on multiple connections

Reviewer 7: This issue is fixed for 1.5.0

Author: Updated

Build system
============

.. rst-class:: v1-4-2 v1-4-1 v1-4-0

NCSDK-6898: Overriding child images
  Adding child image overlay from the :file:`CMakeLists.txt` top-level file located in the :file:`samples` directory overrides the existing child image overlay.

  **Workaround:** Apply the configuration from the overlay to the child image manually.

.. rst-class:: v1-4-2 v1-4-1 v1-4-0

NCSDK-6777: Project out of date when :option:`CONFIG_SECURE_BOOT` is set
  The DFU :file:`.zip` file is regenerated even when no changes are made to the files it depends on.
  As a consequence, SES displays a "Project out of date" message even when the project is not out of date.

  **Workaround:** Apply the fix from `sdk-nrf PR #3241 <https://github.com/nrfconnect/sdk-nrf/pull/3241>`_.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 8: This is fixed by https://github.com/nrfconnect/sdk-nrf/pull/3910 which will be included in the relase, so I guess we can safely remove this.

```suggestion
.. rst-class:: v1-4-2 v1-4-1 v1-4-0
```

Build system
============

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 3: Removed the latest tag.

NCSDK-6898: Overriding child images
  Adding child image overlay from the :file:`CMakeLists.txt` top-level file located in the :file:`samples` directory overrides the existing child image overlay.

  **Workaround:** Apply the configuration from the overlay to the child image manually.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 3: Removed the latest tag.

    Set :option:`CONFIG_USE_DT_CODE_PARTITION` Kconfig option to ``y``.
    Make sure that the code partition is defined and chosen correctly (``offset`` and ``size``).

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0

> Build configuration issues

Reviewer 8: ```suggestion
NCSDK-6117: Build configuration issues
```

This is the JIRA ticket, we might want to add that here.

Reviewer 3: Added JIRA.

  **Workaround:** Execute the flash command from inside the build directory of the child image that is placed on the other core (for example, :file:`build/hci_rpmsg`).

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0

> Secure Partition Manager and application building together

Reviewer 8: ```suggestion
NCSDK-8232: Secure Partition Manager and application building together
```

Reviewer 3: Added JIRA.

  For this reason, do not use :c:macro:`CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING` and :c:macro:`CLOCK_CONTROL_NRF_K32SRC_EXT_LOW_SWING`.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-6362: Do not use the synthesized low frequency clock source
  The synthesized low frequency clock source is neither tested nor intended for usage with MPSL.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-14153: Radio Notification power performance penalty
  The Radio Notification feature has a power performance penalty proportional to the notification distance.
  This means an additional average current consumption of about 600 µA for the duration of the radio notification.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

DRGN-11059: Front-end module API not implemented for SoftDevice Controller
  Front-end module API is currently not implemented for SoftDevice Controller.
  It is only available for 802.15.4.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

> KRKNWK-8842: Low PA or LNA gain when using nRF21540
>   The nRF21540 gain is low or negative in PA or LNA after enabling the feature.
>   The gain can normalize after some time.
>   This issue is described in the errata 4 for nRF21540.

Reviewer 2: As decided, this issue should be rephrased

```suggestion
KRKNWK-8842: MPSL does not support nRF21540 revision 1 or older.
  nRF21540 revision 1 or older and kits that contain this device are not supported by MPSL.
  
  **Workaround:** Check nordicsemi.com for latest information on availability of the product version of nRF21540.
```

Multiprotocol Service Layer (MPSL)
==================================

.. rst-class:: v1-5-0 v1-4-2 v1-4-1

DRGN-15223: `CONFIG_SYSTEM_CLOCK_NO_WAIT` is not supported for nRF5340
  Using :option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` with nRF5340 devices might not work as expected.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1

Reviewer 9: Yeah, it is still not fixed for nrf5340

> DRGN-15176: `CONFIG_SYSTEM_CLOCK_NO_WAIT` is ignored when Low Frequency Clock is started before initializing MPSL

Reviewer 7: @rugeGerritsen or @hurricaneFromMoscow  This one is fixed, right?

Reviewer 3: @rugeGerritsen , @hurricaneFromMoscow , any updates on this?

Reviewer 9: It is fixed in 1.5, yes

  If the application starts the Low Frequency Clock before calling :c:func:`mpsl_init()`, the clock configuration option :option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` has no effect.
  MPSL will wait for the Low Frequency Clock to start.

  **Workaround:** When :option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` is set, do not start the Low Frequency Clock.

> .. rst-class:: v1-5-0 v1-4-0 v1-3-2 v1-3-1 v1-3-0

Reviewer 3: @kromancer , @rugeGerritsen , any updates on this?

> .. rst-class:: v1-5-0 v1-4-0 v1-3-2 v1-3-1 v1-3-0

Reviewer 9: Fixed, yes

DRGN-15064: External Full swing and External Low swing not working
  Even though the MPSL Clock driver accepts a Low Frequency Clock source configuration for External Full swing and External Low swing, the clock control system is not configured correctly.
  For this reason, do not use :c:macro:`CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING` and :c:macro:`CLOCK_CONTROL_NRF_K32SRC_EXT_LOW_SWING`.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-6362: Do not use the synthesized low frequency clock source
  The synthesized low frequency clock source is neither tested nor intended for usage with MPSL.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-14153: Radio Notification power performance penalty
  The Radio Notification feature has a power performance penalty proportional to the notification distance.
  This means an additional average current consumption of about 600 µA for the duration of the radio notification.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 2: Yes, only for 802.15.4. We can rephrase to something like `Front-end modules API not implemented for SoftDevice Controller`, or remove completely from v1.5.0

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 3: Based on conversation with Hubert, rephrased to:
```
.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

DRGN-11059: Front-end module API not implemented for SoftDevice Controller
  Front-end module API is currently not implemented for SoftDevice Controller.
  It is only available for 802.15.4.
```
The JIRA issue was fixed after feature freeze.

Front-end modules API implementation missing
  Front-end modules API is currently not implemented by any protocol.

> 

Reviewer 2: ```suggestion

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

KRKNWK-8842: Low PA or LNA gain when using nRF21540
  nRF21540 gain is low or negative in PA or LNA after enabling the feature. The gain can normalize after some time.
  This issue is described in errata 4 for nRF21540.
  
```

Reviewer 3: Added to the list.

SoftDevice Controller
=====================

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0

Reviewer 3: Removed the latest tag.

DRGN-15226: Link disconnects with reason "LMP Response Timeout (0x22)"
  If the slave receives an encryption request while the "HCI LE Long Term Key Request" event is disabled, the link disconnects with the reason "LMP Response Timeout (0x22)".
  The event is disabled when :option:`CONFIG_BT_SMP` and/or :option:`CONFIG_BT_CTLR_LE_ENC` is disabled.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0

> DRGN-11963: LL control procedures cannot be initiated at the same time

Reviewer 10: DRGN-11963: LL control procedures cannot be initiated at the same time will be fixed in 1.5.0

Reviewer 3: The reference task is still in review, so no change has been made to known issues yet.

Reviewer 10: It is fixed in 1.5.0

  The LL control procedures (LE start encryption and LE connection parameter update) cannot be initiated at the same time or more than once.
  The controller will return an HCI error code "Controller Busy (0x3a)", as per specification's chapter 2.55.

  **Workaround:** Do not initiate these procedures at the same time.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

DRGN-8476: Long packets not supported in connections on Coded PHY
  In connections, the Link Layer payload size is limited to 27 bytes on LE Coded PHY.

.. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

DRGN-9083: AAR populated with zero IRK
  If the application has set an all zeroes IRK for a device in the resolving list, then a resolvable address that can be resolved with the all zeroes IRK will be reported to the application as that device in the advertisement report or the connected event.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

Reviewer 10: DRGN-13921: Directed advertising issues using RPA in TargetA has been fixed

Reviewer 3: Removed the latest tag.

Zephyr
******

> .. rst-class:: v1-5-0
> 
> Disabling USB causes crash
>   Disabling USB by calling :c:func:`usb_disable` with a subsequent call to :c:func:`usb_dc_detach` tries to free unallocated memory and crashes.
> 

Reviewer 11: ```suggestion
```
This issue only affected v1-4-99-dev1

Reviewer 3: Removed from the known issues, as v1.5.0 replaces v1.4.99.

USB Mass Storage Sample Application compilation issues
  :ref:`zephyr:usb_mass` does not compile.

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 8: ```suggestion
.. rst-class:: v1-4-2 v1-4-1 v1-4-0
```
This has been fixed

SEGGER Embedded Studio Nordic Edition
*************************************

> .. rst-class:: v1-5-0 v1-4-2 v1-4-1 v1-4-0

Reviewer 12: The issue below is only valid for SES 5.10d, which is the NCS for 1.4.x.
NCS 1.5.0 ships with SES 5.34a, where this issue is not present.

Reviewer 13: We can handle it in my subsequent known-issues PR. Talk to me @tejlmand (and remove change request) so we can get this PR in
